test/specs/palindrom.websockets.test.js

```diff
@@ -1,509 +1,444 @@
-global.WebSocket = require('mock-socket').WebSocket;
-
-const MockSocketServer = require('mock-socket').Server;
-const Palindrom = require('../../src/palindrom');
-const assert = require('assert');
-const moxios = require('moxios');
-const sinon = require('sinon');
-const {PalindromConnectionError} = require('../../src/palindrom-errors')
-
-describe('Sockets', () => {
-  beforeEach(() => {
-    moxios.install();
-  });
-  afterEach(() => {
-    moxios.uninstall();
-  });
-
-  describe('Palindrom constructor', () => {
-    describe('if `useWebSocket` flag is provided', () => {
-      it('should try to open WebSocket connection ', function(done) {
-        const server = new MockSocketServer('ws://localhost/testURL');
-
-        moxios.stubRequest('http://localhost/testURL', {
-          status: 200,
-          headers: { location: 'http://localhost/testURL' },
-          responseText: '{"hello": "world"}'
+import { Server as MockSocketServer } from 'mock-socket';
+import Palindrom from '../../src/palindrom';
+import assert from 'assert';
+import fetchMock from 'fetch-mock';
+import sinon from 'sinon';
+import { PalindromConnectionError } from '../../src/palindrom-errors';
+import { sleep, getTestURL } from '../utils';
+
+describe('Sockets - if `useWebSocket` flag is provided', () => {
+    let server;
+    afterEach(() => {
+        fetchMock.restore();
+        server.stop();
```

Comment:
Previously we were finishing the test after the `stop` is completed, liek 
```js
          /* stop server async then call done */
          server.stop(done);
```

Should we do it still?, maybe that is the reason of flakiness?

Reply:
Checked that `server.stop` is synchronous.